src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size == threshold) {
+            resize();
+        }
+        if (putVal(key, value, table, getIndex(key, table.length))) {
+            size++;
+        }
```

Comment:
така конструкція не дуже зрозуміла. Адже ми в блоці if зазвичай перевіряємо якусь умову і на основі її виконуємо лоіку, а зараз виходить що в умові if ми вставляємо елемент. Щоб це виглядало більш логіно, краще трішки відформатувати таку конструкцію:
```suggestion
        boolean wasNewNodeCreated = putVal(key, value, table, getIndex(key, table.length));
        if (wasNewNodeCreated) {
            size++;
        }
```